tests/phpunit/tests/query/searchColumns.php

```diff
@@ -0,0 +1,381 @@
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertStringContainsString( 'post_title', $q->request );
+		$this->assertStringContainsString( 'post_excerpt', $q->request );
+		$this->assertStringContainsString( 'post_content', $q->request );
+		$this->assertEqualSets( array( self::$pid1, self::$pid2, self::$pid3 ), $q->posts );
+	}
+
+	/**
+	 * The search should support ignore a non-existing search column when used together with a supported one.
+	 */
+	public function test_s_should_ignore_non_existing_search_column_when_used_with_supported_one() {
+		$q = new WP_Query(
+			array(
+				's'              => 'foo',
+				'search_columns' => array( 'post_title', 'post_non_existing_column' ),
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertEqualSets( array( self::$pid1 ), $q->posts );
+	}
+
+	/**
+	 * The search should support search columns when searching multiple terms.
+	 */
+	public function test_s_should_support_search_columns_when_searching_multiple_terms() {
+		$q = new WP_Query(
+			array(
+				's'              => 'foo bar',
+				'search_columns' => array( 'post_content' ),
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertEqualSets( array( self::$pid2, self::$pid3 ), $q->posts );
+	}
+
+	/**
+	 * The search should support search columns when searching for a sentence.
+	 */
+	public function test_s_should_support_search_columns_when_sentence_true() {
+		$q = new WP_Query(
+			array(
+				's'              => 'bar foo',
+				'search_columns' => array( 'post_content' ),
+				'sentence'       => true,
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertEqualSets( array( self::$pid3 ), $q->posts );
+	}
+
+	/**
+	 * The search should support search columns when searching for a sentence.
+	 */
+	public function test_s_should_support_search_columns_when_sentence_false() {
+		$q = new WP_Query(
+			array(
+				's'              => 'bar foo',
+				'search_columns' => array( 'post_content' ),
+				'sentence'       => false,
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertEqualSets( array( self::$pid2, self::$pid3 ), $q->posts );
+	}
+
+	/**
+	 * The search should support search columns when searched with a term exclusion.
+	 */
+	public function test_s_should_support_search_columns_when_searched_with_term_exclusion() {
+		$q = new WP_Query(
+			array(
+				's'              => 'bar -baz',
+				'search_columns' => array( 'post_excerpt', 'post_content' ),
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertEqualSets( array( self::$pid2 ), $q->posts );
+	}
+
+	/**
+	 * The search columns should be filterable with the `post_search_columns` filter.
+	 */
+	public function test_search_columns_should_be_filterable() {
+		add_filter( 'post_search_columns', array( $this, 'post_supported_search_column' ), 10, 3 );
+		$q = new WP_Query(
+			array(
+				's'      => 'foo',
+				'fields' => 'ids',
+			)
+		);
+		remove_filter( 'post_search_columns', array( $this, 'post_supported_search_column' ) );
```

Comment:
```suggestion
```
Filters are automatically removed between tests, so no need to remove it here.

🔢 Applies to the other two instances in the PR.

Reply:
I didn't know that, thanks !